### PR TITLE
chore: use java 21 in samples and templates

### PR DESCRIPTION
**Pull request by octonato**

### CODE REVIEW

**Comment by octonato** [comment]
@aludwiko, re-work of #1905 

I took the path of redefining `java.version` property. Seems more future proof in the sense that whatever the spring parent pom defines, we should be able to pin it to a version that we support.

**Comment by aludwiko** [comment]
Some issues with:
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project java-spring-eventsourced-shopping-cart: Fatal error compiling: error: release version 21 not supported -> [Help 1]
```

I would also introduce some actual java 21 change, just to make sure that everything works. You could for instance change this:
```
  @EventHandler
  public User onEvent(User.UserWasCreated evt) {
    return User.onEvent(evt);
  }

  @EventHandler
  public User onEvent(User.EmailAssigned evt) {
    return currentState().onEvent(evt);
  }


  @EventHandler
  public User onEvent(User.EmailUnassigned evt) {
    return currentState();
  }
  
  ```
 to
 ```
   @EventHandler
  public User onEvent(User evt) {
    switch (evt)
       ...
  }
  ```
in `java-spring-choreography-saga-quickstart`

**Comment by octonato** [comment]
for the record, failing on CI because CI is using jdk17

File: samples/java-protobuf-customer-registry-views-quickstart/pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
     <maven.build.timestamp.format>yyyyMMddHHmmss</maven.build.timestamp.format>
     <mainClass>customer.Main</mainClass>
 
-    <jdk.target>11</jdk.target>
+    <java.version>21</java.version>
```
[the remark this answers is not in the document]
Comment (octonato) [reply samples/java-protobuf-customer-registry-views-quickstart/pom.xml:21]: seems to work, but there is still a major thing to do

The images are on adoptopenjdk 11. We need to move to 21 as well.

File: maven-java/kalix-maven-archetype-event-sourced-entity/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -132,7 +132,7 @@
               <name>${D}{dockerImage}:%l</name>
               <build>
                 <!-- Base Docker image which contains jre-->
-                <from>docker.io/library/adoptopenjdk:${D}{jdk.target}-jre-hotspot</from>
```
Comment (octonato) [inline maven-java/kalix-maven-archetype-event-sourced-entity/src/main/resources/archetype-resources/pom.xml:135]: We were previously using the jdk version here, but it should pin to fixed version.

File: samples/java-protobuf-customer-registry-views-quickstart/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (octonato) [reply samples/java-protobuf-customer-registry-views-quickstart/pom.xml:21]: @aludwiko, I think the issue was related with the build. We build the sdk using jdk11 (protobuf) or jdk17 (code-first). 

I'm not touching it as users should be able to choose 11 or 17 if they prefer for some reason. 

We are only touching the samples and templates for now.

[the remark this answers is not in the document]
Comment (aludwiko) [reply samples/java-protobuf-customer-registry-views-quickstart/pom.xml:21]: Ok, I get it now.

**Review by aludwiko — APPROVED** [review]
LGTM

**Pull request merged**